python/amici/sbml_import.py

```diff
@@ -492,6 +557,141 @@ def get_species_initial(index, conc):
              for specie in species
         ])
 
+        # Process assignment and rate rules for species and compartments.
+        # Compartments with rate rules are implemented as species. Species and
+        # compartments with assignments are implemented as observables (and
+        # replaced with their assignment in all expressions). Note that, in the
+        # case of species, rate rules may describe the change in amount, not
+        # concentration, of a species.
+        rules = self.sbml.getListOfRules()
+        compartmentvars = self.compartment_symbols.free_symbols
+        # compartments with rules are replaced with constants in the relevant
+        # equations during the _replace_in_all_expressions call inside
+        # _process_rules
+        for rule in rules:
+            ### Taken from _process_rules
+            if rule.getFormula() == '':
+                continue
+            variable = sp.sympify(rule.getVariable(),
+                                  locals=self.local_symbols)
+            # avoid incorrect parsing of pow(x, -1) in symengine
+            formula = sp.sympify(_parse_logical_operators(
+                sbml.formulaToL3String(rule.getMath())),
+                locals=self.local_symbols)
+            formula = _parse_special_functions(formula)
+            _check_unsupported_functions(formula, 'Rule')
+            ###
+
+            # Species rules are processed first, to avoid processing
+            # compartments twice (as compartments with rate rules are
+            # implemented as species). Could also be avoided with a
+            # `not in self.compartment_rate_rules` condition.
+            if variable in self.symbols['species']['identifier']:
+                if rule.getTypeCode() == sbml.SBML_ASSIGNMENT_RULE:
+                    # Handled in _process_rules and _process_observables.
+                    # TODO: set a species initial value to its assignment rule?
```

Comment:
setting initial value shouldn't be necessary here, as its defined by the assignment rule.